simulation_ws/src/rl-agent/markov/environments/mars_env.py

```diff
@@ -129,6 +130,9 @@ def __init__(self):
         rospy.Subscriber('/camera/image_raw', sensor_image, self.callback_image)
         # IMU Sensors
         rospy.Subscriber('/imu/wheel_lb', Imu, self.callback_wheel_lb)
+        self.writer = SummaryWriter()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *writer*. After you use this resource, use this method to prevent a resource leak.